### PR TITLE
Patterns API: Add 'inserter' to the schema

### LINKED ISSUE #40414: Patterns with `Inserter: false` are visible in Inserter with Gutenberg 13.0
### Description

Prior to Gutenberg 13.0, you could register patterns with `'inserter' => false,` in JS or the `Inserter: false` comment in PHP (the new method in 12.9) and the pattern would not be visible in the inserter. In 13.0 this no longer works. My guess is this has to do with some recent changes to the Patterns API, but not sure.

### Step-by-step reproduction instructions

1. Make sure 13.0+ is installed
2. Install the Twenty Twenty-Two theme
3. Open the Pattern Explorer and you will see an "Uncategorized" section
4. This section will contain the patterns that should be hidden. 

### Screenshots, screen recording, code snippet

Patterns Visible in the Editor"
image

Example 404 pattern with `'inserter' => false`:
image

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Fixes #40414.

PR adds the missing `inserter` field to the Patterns item schema.

## Why?
Currently, this field is omitted from the response, and patterns that should be hidden in the inserter are visible.

## Testing Instructions
1. Activate the TT2 theme.
2. Open a post editor.
3. Open the Pattern Explorer.
4. Confirm that the "Uncategorized" section isn't visible.

### CODE REVIEW

**Comment by ndiego** [comment]
Thanks for this @Mamaduka! This PR fixes the root issue, both for patterns registered via JS and PHP in the `/patterns` folder. I have approved assuming any relevant failed tests are resolved. The two that are currently failing appear unrelated.

**Review by ndiego — APPROVED** [review]

**Comment by ndiego** [comment]
@Mamaduka, once merged, let's add the backport tag. This should get in 6.0.

**Comment by Mamaduka** [comment]
Thanks, @ndiego. I think those are flaky tests.

**Comment by Mamaduka** [comment]
@adamziel, @gziolo, here's WP core PR - https://github.com/WordPress/wordpress-develop/pull/2600.

**Comment by gziolo** [comment]
> @adamziel, @gziolo, here's WP core PR - [WordPress/wordpress-develop#2600](https://github.com/WordPress/wordpress-develop/pull/2600).

Actually, there is nothing to do about this PR in the context of WordPress core. Let's add the linked PR in core to the project.

**Pull request merged**
